src/main/java/g0301_0400/s0354_russian_doll_envelopes/Solution.java

```diff
@@ -0,0 +1,32 @@
+package g0301_0400.s0354_russian_doll_envelopes;
+
+// #Hard #Array #Dynamic_Programming #Sorting #Binary_Search
+
+import java.util.Arrays;
+
+public class Solution {
+    public int maxEnvelopes(int[][] envelopes) {
+        Arrays.sort(envelopes, (a, b) -> a[0] != b[0] ? a[0] - b[0] : b[1] - a[1]);
+        int[] tails = new int[envelopes.length];
+        int size = 0;
+        for (int[] enve : envelopes) {
+            int i = 0;
+            int j = size;
+
+            while (i != j) {
+                int mid = i + ((j - i) >> 1);
+
```

Comment:
Please remove empty lines

Reply:
I usually code spaces like that to be easier to see than just writing a block

Reply:
The idea to use plugin to format codes. If it allows to remove empty line - remove it.